### PR TITLE
PIC-2031 - PIC-2031-add-phone-to-defendant

### PR DESCRIPTION
**Pull request by rschallapalli**

### CODE REVIEW

File: src/main/java/uk/gov/justice/probation/courtcaseservice/controller/mapper/PhoneNumberMapper.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package uk.gov.justice.probation.courtcaseservice.controller.mapper;
+
+import lombok.experimental.UtilityClass;
+import uk.gov.justice.probation.courtcaseservice.controller.model.PhoneNumber;
+import uk.gov.justice.probation.courtcaseservice.jpa.entity.PhoneNumberEntity;
+
+import java.util.Optional;
+
+@UtilityClass
+public class PhoneNumberMapper {
```
Comment (jhevans) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/controller/mapper/PhoneNumberMapper.java:10]: Can't see unit tests for this

File: data_dictionary/Dockerfile [hunk 2]
```diff
@@ -5,6 +5,6 @@ RUN chown -R nginx:nginx /var/cache/nginx
 RUN chown -R nginx:nginx /var/log/nginx
 RUN touch /var/run/nginx.pid && chown -R nginx:nginx /var/run/nginx.pid
 
-RUN apk -U upgrade
+RUN apt update --nocache
```
Comment (jhevans) [inline data_dictionary/Dockerfile:8]: I think `apt update` only updates the packages lists, you need to do a `apt upgrade` to install the new packages

File: src/main/java/uk/gov/justice/probation/courtcaseservice/controller/mapper/PhoneNumberMapper.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    public static PhoneNumber mapFrom(final PhoneNumberEntity phoneNumberEntity) {
```
[the remark this answers is not in the document]
Comment (rschallapalli) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/controller/mapper/PhoneNumberMapper.java:12]: Definitely agree with all these points, thanks

File: src/main/java/uk/gov/justice/probation/courtcaseservice/controller/mapper/PhoneNumberMapper.java (hunk 1 above)
Comment (rschallapalli) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/controller/mapper/PhoneNumberMapper.java]: moved to DTO now with tests

File: data_dictionary/Dockerfile (hunk 2 above)
Comment (rschallapalli) [reply data_dictionary/Dockerfile]: done

File: .snyk [hunk 4]
```diff
@@ -9,5 +9,9 @@ ignore:
     - '*':
         reason: Believed we don't used the open SSL libraries. Will revisit
         expires: 2022-03-04T00:00:00.000Z
+  SNYK-ORACLE7-EXPAT-2427672:
+    - '*':
+        reason: Hoping this will be patched soon
```
Comment (jhevans) [inline .snyk:14]: Can I request we re-word this, I feel like 'hope' doesn't read well as a basis for security decisions 😅 

```suggestion
        reason: Does not appear to be exploitable, accept risk in short term pending patch
```

File: src/test/java/uk/gov/justice/probation/courtcaseservice/controller/model/PhoneNumberTest.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.justice.probation.courtcaseservice.jpa.entity.EntityHelper.DEFENDANT_PHONE_NUMBER;
+import static uk.gov.justice.probation.courtcaseservice.jpa.entity.EntityHelper.DEFENDANT_PHONE_NUMBER_ENTITY;
+
+class PhoneNumberTest {
+
+    @Test
+    void shouldMapToEntityOnAsEntity() {
+        assertThat(DEFENDANT_PHONE_NUMBER.asEntity()).isEqualTo(DEFENDANT_PHONE_NUMBER_ENTITY);
```
Comment (jhevans) [inline src/test/java/uk/gov/justice/probation/courtcaseservice/controller/model/PhoneNumberTest.java:13]: Nice bit of re-use this, very concise!

**Review by jhevans — APPROVED** [review]
Looks good 👍

**Pull request merged**